### ISSUE TITLE
jsx-wrap-multiline false positives

### ISSUE THREAD

**Issue by paranoidjk** [palantir/tslint-react#79]
I think the case below should not cause error:

https://github.com/ant-design/ant-design-mobile/blob/master/components/accordion/demo/basic.tsx#L13
 
```bash
components/accordion/demo/basic.tsx[13, 9]: Multiline JSX elements must be wrapped in parentheses
```

```js
// bad
const button = <button type="submit">
    Submit
</button>;
// good
const button = (
    <button type="submit">
        Submit
    </button>
);

// why this is bad ? any reason we need wrap inner button with parentheses?
const button = (
   <div>
       <button type="submit">
          Submit
      </button>
   <div>
);
```

**Comment by adidahiya** [comment]
I can repro this, looks like a bug

**Comment by adidahiya** [comment]
more false positives I've found:

```
return [
    <button type="submit">
        Submit
    </button>,
    <button type="button">
        Cancel
    </button>
];
```

**Comment by IllusionMH** [comment]
Is it still can be reproduced?
Checked by replacing `test/rules/jsx-wrap-multiline/test.tsx.lint` with all code examples from this issue and link in original post. Win 10, tslint 4.5.1, tsc 2.2.1

Only example of bad code triggers error. Example with button in div passes lint as well as JSX Elements in array.

@adidahiya example from your last comment is false positive or false negative? I'd expect braces there too, but no lint errors at this moment.

**Comment by giladgray** [comment]
the array false positive @adidahiya mentioned above has forced me to disable this rule... we use that pattern everywhere.

**Comment by QuantumInformation** [comment]
I'm getting Multiline JSX elements must be wrapped in parentheses

for 
![image](https://user-images.githubusercontent.com/216566/33025525-27abd756-ce06-11e7-95f9-8076c972ac43.png)

**Comment by adidahiya** [comment]
@QuantumInformation yeah, that's the same as my comment above https://github.com/palantir/tslint-react/issues/79#issuecomment-289923337

**Comment by wichert** [comment]
Another repro using React 16.2:

```
return (
  <>
    <div>One</div>
    <div>Two</div>
  </>
)
```

**Comment by rhys-vdw** [comment]
I wouldn't call this a false positive:

```tsx
return [
    <button type="submit">
        Submit
    </button>,
    <button type="button">
        Cancel
    </button>
];
```

I prefer:

```tsx
return [(
    <button type="submit">
        Submit
    </button>
), (
    <button type="button">
        Cancel
    </button>
)];

One reason is that when refactoring it's very easy to leave the trailing `,` after the closing tag, which can end up appearing in your rendered markup.

**Comment by Primajin** [comment]
@rhys-vdw problem with this way is that it is contradicting ESLint / Prettier rules...

**Comment by evenfrost** [comment]
```tsx
render(
  <ApolloProvider client={client}>
    <Router history={history}>
      <App />
    </Router>
  </ApolloProvider>,
  document.querySelector('main') as HTMLElement,
);
```

Here's the code that throws false positive `jsx-wrap-multiline` for me.

**Comment by adidahiya** [comment]
I don't think we're going to fix this in tslint at this point. See https://github.com/palantir/tslint-react/issues/210 and https://github.com/palantir/tslint/issues/4534. You're better off using prettier.

**Comment by dawnmist** [comment]
Problem is that the 4.0.0 release of tslint-react no longer permits the standard prettier formatting for the reactDom.render() example, whereas they worked fine together in tslint-react 3.6.0.

Prettier removes any brackets around the JSX in the first parameter for reactDom.render, so it is not possible for:
```js
 render(
  <ApolloProvider client={client}>
    <Router history={history}>
      <App />
    </Router>
  </ApolloProvider>,
  document.querySelector('main') as HTMLElement,
);
```
to be written as
```js
render((
    <ApolloProvider client={client}>
      <Router history={history}>
        <App />
      </Router>
    </ApolloProvider>
  ),  document.querySelector('main') as HTMLElement,
);
```
when using prettier like tslint 4.0.0 now wants it to be. Can whatever was changed between those two versions for this particular rule & case please be reverted?

**Comment by adidahiya** [comment]
but why would you want to enable `jsx-wrap-multiline` when `prettier` is enabled?

**Comment by dawnmist** [comment]
Good question, and one I really hadn't thought a lot about previously because the two packages worked in harmony previously and now do not.

Flipping the question back - if you're trying to encourage people to use prettier instead of tslint-react, why would you introduce conflicting formatting that makes migration between the two projects require additional work when it had previously been working in harmony? Now anyone using tslint-react has had to change their render functions from a prettier-compatible format into a prettier-incompatible format, only to have them changed back again after migrating to prettier. That introduces unnecessary noise to git commit/blame histories for no real benefit, and no good reason when the change wasn't needed in the first place.

**Comment by adidahiya** [comment]
tslint-react had this rule before prettier was widely used in the typescript ecosystem and before we started using prettier in Palantir projects. all the code formatting rules in tslint & tslint-react are "legacy" rules in that sense; they are not guaranteed to play nicely with prettier. if you use [tslint-config-prettier](https://github.com/prettier/tslint-config-prettier) (the recommended configuration approach), you should have all formatting rules disabled in favor of prettier.
